k4FWCore/components/PodioInput.cpp

```diff
@@ -21,21 +21,172 @@
 
 #include "k4FWCore/PodioDataSvc.h"
 
+#include "edm4hep/MCParticleCollection.h"
```

Comment:
The problem is that we have to register objects with the type we want to get, otherwise there is the check I have mentioned in one of the comments above that will try to dynamic_cast whatever is in the store (`DataWrapper<collectionType>`) to what we are asking for. Pushing a `DataWrapper<podio::CollectionBase>` like before without templating the registration and getting back a `AnyDataWrapper<class>` doesn't work since you can't dynamic_cast one into the other